build.sh

```diff
@@ -221,20 +225,7 @@ esac
 __BuildOS=$__HostOS
 __BuildType=Debug
 __CMakeArgs=DEBUG
-
-case $__HostOS in
-    FreeBSD)
-        __monoroot=/usr/local
-        ;;
-    OSX)
-        __monoroot=/Library/Frameworks/Mono.framework/Versions/Current
-        ;;
-    *)
-        __monoroot=/usr
-        ;;
-esac
-
-__referenceassemblyroot=$__monoroot/lib/mono/xbuild-frameworks
+__referenceassemblyroot="$(csharp FindMonoXBuildFrameworks.cs)"
```

Comment:
@stephentoub I could not find a way to determine the GAC location reliably via regular scripting. Mono has some environment variables that *might* be there aren't guaranteed.

I understand the use of hardcoding the path, but the problem is that this path isn't guaranteed; this will pick up the changes to whatever Mono the user is using by default since it uses the same algorithm that the terminal uses to find what the user means by typing ```mono```. I think it would be more confusing to have this hardcoded and require everyone to have the exact same configuration rather than follow the user preference and functionality set by the path; ie when you run ```mcs``` or ```mono``` it will use a certain PCL and we will adhere to that. It seems like if you have multiple versions installed and this path hardcoded, it becomes a debugging issue to figure out which version CoreFX used and which version your test program used; I'm not tied to either way of doing this, I was simply trying to make it fit with how ```which``` works and how the system will find the PCL.